subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultFileCollectionDependency.java

```diff
@@ -0,0 +1,140 @@
+        return source.equals(selfResolvingDependency.source);
+    }
+
+    @Override
+    public DefaultFileCollectionDependency copy() {
+        return new DefaultFileCollectionDependency(targetComponentId, source);
+    }
+
+    @Nullable
+    @Override
+    public ComponentIdentifier getTargetComponentId() {
+        return targetComponentId;
+    }
+
+    @Override
+    public String getGroup() {
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return "unspecified";
+    }
+
+    @Override
+    public String getVersion() {
+        return null;
+    }
+
+    @Override
+    @Deprecated
+    @SuppressWarnings("deprecation")
+    public void resolve(org.gradle.api.internal.artifacts.CachingDependencyResolveContext context) {
+        context.add(source);
+    }
+
+    @Override
+    @Deprecated
+    public Set<File> resolve() {
+        if (emitDeprecation()) {
+            DeprecationLogger.deprecate("Directly resolving a file collection dependency's files")
+                .withAdvice("Add the dependency to a resolvable configuration and resolve the configuration.")
+                .willBecomeAnErrorInGradle9()
+                .withUpgradeGuideSection(8, "deprecate_self_resolving_dependency")
+                .nagUser();
+        }
+
+        return source.getFiles();
+    }
+
+    @Override
+    @Deprecated
+    public Set<File> resolve(boolean transitive) {
+
+        DeprecationLogger.deprecate("Directly resolving a file collection dependency's files")
+            .withAdvice("Add the dependency to a resolvable configuration and resolve the configuration.")
+            .willBecomeAnErrorInGradle9()
+            .withUpgradeGuideSection(8, "deprecate_self_resolving_dependency")
+            .nagUser();
+
+        return source.getFiles();
+    }
+
+    @Override
+    @Deprecated
+    public TaskDependency getBuildDependencies() {
+        if (emitDeprecation()) {
+            DeprecationLogger.deprecate("Accessing the build dependencies of a file collection dependency")
+                .withAdvice("Add the dependency to a resolvable configuration use the configuration to track task dependencies.")
+                .willBecomeAnErrorInGradle9()
+                .withUpgradeGuideSection(8, "deprecate_self_resolving_dependency")
+                .nagUser();
+        }
+
+        return source.getBuildDependencies();
+    }
+
+    private static boolean emitDeprecation() {
+        // TODO #27437: Always emit deprecation here.
+        // AGP currently uses these deprecated APIs. An issue has been filed for them to migrate away.
+        // This property exists to allow AGP to test whether they have properly resolved this deprecation.
+        // This property WILL be removed without warning.
+        return System.getProperty("org.gradle.internal.deprecation.preliminary.DefaultFileCollectionDependency.enabled", "false").equals("true");
```

Comment:
`Boolean.getBoolean(...)` does what you want

Reply:
Changed